### PR TITLE
Add guiding principles doc

### PR DESCRIPTION
**Pull request by MichaelSimons**
I tried to capture the set of guiding principles being used when authoring/maintaining the .NET Core Docker images.  They establish a framework for the behavior customers can expect and the decision-making process when making changes to the images.

### CODE REVIEW

**Comment by MichaelSimons** [comment]
@mthalman - I'd appreciate your feedback before I sending this to a wider audience.

File: documentation/guiding-principles.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+1. Breaking changes are not allowed within a release.  This includes changes such as adding/removing components, adding/removing ENVs, and major/minor version changes to included components.
+
+1. There should be parity in the matrix of supported image variants.  Examples include:
+    1. If support for a new distro is added for a particular .NET Core version, then new `runtime-deps`, `runtime`, `aspnet`, and `sdk` images should all be added for the new distro.
+    1. If a new component is added, it should be available within all of the supported architectures - `amd64`, `arm32`, `arm64`, etc.
+
+1. The set of supported Linux distros and versions will evolve with the ecosystem.
+
+    **// Question: Should something be stated on how we uptake new OS versions?**
```
Comment (mthalman) [inline documentation/guiding-principles.md]: I think it makes sense to have a general statement about supporting distros as you've done.  As for the details of that, I think it belongs on a separate document.  I would say we should take Rich's content from [here](https://devblogs.microsoft.com/dotnet/using-net-and-docker-together-dockercon-2019-update/#user-content-platform-matrix-and-support) and put it into its own document which can be linked from here.

File: documentation/guiding-principles.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+1. Nano Server is the only Windows SKU supported.  Nano Server is the best Windows SKU to run .NET Core apps from a performance perspective.  There are .NET Core scenarios which are not supported by Nano Server.  For these cases, customers will need to manage their own custom images based on [Windows Server Core](https://hub.docker.com/_/microsoft-windows-servercore) or [ Windows](https://hub.docker.com/_/microsoft-windows).
+
+## Image Tagging
+
+The .NET Core image tags strive to align with the tagging practices utilized by the [Official Images on Docker Hub](https://hub.docker.com/search?q=&type=image&image_filter=official).
```
Comment (mthalman) [inline documentation/guiding-principles.md]: I think the intent of this doc should just be the high-level set of principles.  So this statement is good here.  But the content below gets into the nitty gritty details of how the tag names work.  This should be in a separate document I think.

File: documentation/guiding-principles.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+* `<Major.Minor Version>` - The `Major.Minor` number of the .NET Core version included in the image.  The tag is updated to always reference the most recent patch that is currently available for the `Major.Minor` release.
+
+* `<OS>` - The name of the OS release and variant the image is based.  The image the tag references is updated whenever a new OS patch is released.  The OS release name does support pinning to a specific OS patch.  If OS patch pinning is required then the image digest should be used (e.g. `mcr.microsoft.com/dotnet/core/runtime@sha256:fff4cfe761fde9f3b72377e350eda7cd82caf0c1ec6be281b92d8614860fa449`).
+
+* `<Architecture>` - The architecture the image is based on.  `amd64` is the implied default if no architecture is specified.
+
+## Engineering
+
+1. Images will be included as part of the .NET Core release process.  This means the Docker images will be release at the same time as the core product.
```
Comment (mthalman) [inline documentation/guiding-principles.md]:
```suggestion
1. Images will be included as part of the .NET Core release process.  This means the Docker images will be release at the same time as the core product.
```
```suggestion
1. Images will be included as part of the .NET Core release process.  This means the Docker images will be released at the same time as the core product.
```

File: documentation/guiding-principles.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+1. Images will get rebuilt within hours of base image changes. For example suppose a particular version of Alpine is patched.  The .NET Core images based on this version of Alpine will get rebuilt based on this new base image within hours of its release.
```
Comment (mthalman) [inline documentation/guiding-principles.md]:
```suggestion
1. Images will be rebuilt within hours of base image changes. For example, suppose a particular version of Alpine is patched.  The .NET Core images based on this version of Alpine will be rebuilt based on this new base image within hours of its release.
```

**Comment by mthalman** [comment]
Not sure if this would be considered a principle that we're following but thought I would mention it: production builds make use of stable tools (i.e. don't make use of experimental features in Docker).

File: documentation/guiding-principles.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+1. Components installed within the images are required to have the same or longer support lifecycle as [.NET Core](https://dotnet.microsoft.com/platform/support/policy/dotnet-core).  For example, if a component is included with an LTSC release of .NET, then that component version will need to be supported for the life of the LTSC release.  Components are expected to be patched as appropriate.
+
+1. Breaking changes are not allowed within a release.  This includes changes such as adding/removing components, adding/removing ENVs, and major/minor version changes to included components.
+
+1. There should be parity in the matrix of supported image variants.  Examples include:
+    1. If support for a new distro is added for a particular .NET Core version, then new `runtime-deps`, `runtime`, `aspnet`, and `sdk` images should all be added for the new distro.
+    1. If a new component is added, it should be available across all supported OS types and architectures.
+
+1. The matrix of supported container operating systems and versions will evolve on a continuous bases. The [Supported Platforms](supported-platforms.md) describes this in detail.
```
Comment (mthalman) [inline documentation/guiding-principles.md]:
```suggestion
1. The matrix of supported container operating systems and versions will evolve on a continuous basis. The [Supported Platforms](supported-platforms.md) describes this in detail.
```

File: documentation/supported-platforms.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# Supported Platforms
+
+This document describes the platfrom matrix supported by the .NET Core Docker repositories.
+
+## Operating Systems
+
+.NET Core tries to [support a broad set of operating systems and versions](https://github.com/dotnet/core/blob/master/os-lifecycle-policy.md). With containers, it’s too expensive and confusing to support the full matrix of options. In practice, images are produced for select set of distros.
```
Comment (mthalman) [inline documentation/supported-platforms.md]:
```suggestion
.NET Core tries to [support a broad set of operating systems and versions](https://github.com/dotnet/core/blob/master/os-lifecycle-policy.md). With containers, it’s too expensive and confusing to support the full matrix of options. In practice, images are produced for a select set of distros.
```

File: documentation/supported-platforms.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+### Linux
+
+Each distribution has a unique approach to releasing, schedule and end-of life (EOL). This prohibits the definition of a one-size-fits-all policy. Instead, a policy is defined for each supported distro.
+
+- Alpine — support tip and retain support for one quarter (3 months) after a new version is released.
+- Debian — support one Debian version per .NET Core version, whichever Debian version is the latest when a given .NET Core version ships. This is also the default Linux image used for a given multi-arch tag.
+- Ubuntu — support one Ubuntu version per .NET Core version, whichever Ubuntu version is the latest LTS version when a given .NET Core version ships.  In addition, as it gets closer to new Ubuntu LTS versions, support will be added for non-LTS Ubuntu versions a means of validating the new LTS versions.
```
Comment (mthalman) [inline documentation/supported-platforms.md]:
```suggestion
- Ubuntu — support one Ubuntu version per .NET Core version, whichever Ubuntu version is the latest LTS version when a given .NET Core version ships.  In addition, as it gets closer to new Ubuntu LTS versions, support will be added for non-LTS Ubuntu versions as a means of validating the new LTS versions.
```

File: documentation/tagging-guidelines.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+    1. Include Windows entries for each supported version.
+
+## Tag Parts
+
+* `<Major.Minor.Patch Version>` - The `Major.Minor.Patch` number of the .NET Core version included in the image.
+
+* `<Major.Minor Version>` - The `Major.Minor` number of the .NET Core version included in the image.  The tag is updated to always reference the most recent patch that is currently available for the `Major.Minor` release.
+
+* `<OS>` - The name of the OS release and variant the image is based.  The image the tag references is updated whenever a new OS patch is released.  The OS release name does support pinning to specific OS patches.  If OS patch pinning is required then the image digest should be used (e.g. `mcr.microsoft.com/dotnet/core/runtime@sha256:fff4cfe761fde9f3b72377e350eda7cd82caf0c1ec6be281b92d8614860fa449`).
```
Comment (mthalman) [inline documentation/tagging-guidelines.md]:
```suggestion
* `<OS>` - The name of the OS release and variant the image is based upon.  The image the tag references is updated whenever a new OS patch is released.  The OS release name does support pinning to specific OS patches.  If OS patch pinning is required then the image digest should be used (e.g. `mcr.microsoft.com/dotnet/core/runtime@sha256:fff4cfe761fde9f3b72377e350eda7cd82caf0c1ec6be281b92d8614860fa449`).
```

File: documentation/guiding-principles.md (hunk 1 above)
Comment (MichaelSimons) [reply documentation/guiding-principles.md]: I agree - done.

File: documentation/guiding-principles.md (hunk 2 above)
Comment (MichaelSimons) [reply documentation/guiding-principles.md]: I agree - done.

File: documentation/tagging-guidelines.md [hunk 9]
```diff
@@ -1,28 +1,58 @@
...
+
+    * `2.1`
+    * `2.2`
+    * `3.0`
+    * `3.1`
+
 1. `latest` - Will always refer to the most recent non-prerelease `<Major.Minor.Patch Version>` image.
 
-    All shared tags [support multiple platforms](https://blog.docker.com/2017/09/docker-official-images-now-multi-platform/) and have the following characteristics:
+All shared tags [support multiple platforms](https://blog.docker.com/2017/09/docker-official-images-now-multi-platform/) have the following characteristics:
```
Comment (mthalman) [inline documentation/tagging-guidelines.md]: You remove the word `and` here but I think it's still needed.

**Pull request merged**